pkgs/build-support/dlang/dub-to-nix/dub-to-nix.sh

```diff
@@ -0,0 +1,43 @@
+tmp=$(realpath "$(mktemp -td dub-to-nix.XXXXXX)")
+trap 'rm -r "$tmp"' EXIT
+
+if [ ! -f dub.selections.json ]; then
+    echo "The file \`dub.selections.json\` does not exist" >&2
+    echo "run \`dub upgrade --annotate\` to generate it" >&2
+    exit 1
+fi
+
+# Create space separated "pname version" pair for each dependency
+deps=$(jq -r '.versions | to_entries[] | "\(.key) \(.value)"' dub.selections.json)
+
+IFS=$'\n'
+for dep in $deps; do
+    IFS=" " read -r pname version <<< "$dep"
+    if [[ $version == "~"* ]] ; then
+        echo "Package \"$pname\" has a branch-type version \"$version\", which doesn't point to a fixed version" >&2
+        echo "You can resolve it by manually changing the required version to a fixed one inside \`dub.selections.json\`" >&2
+        echo "When packaging, you might need to create a patch for \`dub.sdl\` or \`dub.json\` to accept the changed version" >&2
+        exit 1
+    fi
+done
+
+echo "{ makeDubDep }:"
+echo "["
```

Comment:
current approach is fine but maybe it's less fragile to output a json file with just data (pname, version, sha256), and then move the logic in buildDubPackage to:
-  load the json with builtins.fromJSON and builtins.readFile
- map makeDubDep on it

also consider running shellcheck on this script

Reply:
I like this idea. Essentially, it would just be a better `dub.selections.json`. Time to do some more `jq` magic.

Reply:
Instead of doing jq magic you could also write the script in a language that supports json out of the box like python ;)

Reply:
I did it in python :)
I'm not sure what's the cleanest way of turning it into an executable, though.